src/S3/S3Client.php

```diff
@@ -219,6 +219,8 @@ public static function isBucketDnsCompatible($bucket)
 
     public function createPresignedRequest(CommandInterface $command, $expires)
     {
+        $command->getHandlerList()->remove('signer');
```

Comment:
Does it matter that we've mutated the command in this method? Maybe we can clone the command or, a more involved change, we could return the removed handler when calling remove, and add it back at the end of the method.